vmdb/app/controllers/application_controller.rb

```diff
@@ -2743,4 +2745,20 @@ def set_gettext_locale
     end
     FastGettext.set_locale(locale)
   end
+
+  def get_session_data
+    @title      = ui_lookup(:tables => self.class.table_name)
+    @layout     = self.class.table_name
+    prefix      = self.class.session_key_prefix
+    @lastaction = session["#{prefix}_lastaction".to_sym]
+    @showtype   = session["#{prefix}_showtype".to_sym]
+    @display    = session["#{prefix}_display".to_sym]
+  end
+
+  def set_session_data
+    prefix                                 = self.class.session_key_prefix
+    session["#{prefix}_lastaction".to_sym] = @lastaction
+    session["#{prefix}_showtype".to_sym]   = @showtype
+    session["#{prefix}_display".to_sym]    = @display unless @display.nil?
+  end
```

Comment:
This is definitely not appropriate for the Application controller. Only put things in the application controller that are appropriate for most controllers to generally do.  Please move this code to a module and mix that module in to your new controllers.